File: frontend/src/app/core/services/pago.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import {
  CanjeBancarioRequest,
  IngresoExternoRequest,
  ProcesarPagoRequest,
  ProcesarPagoResponse,
  Recibo,
  TipoRecibo,
} from '../models/recibo.model';

@Injectable({ providedIn: 'root' })
export class PagoService {
  private readonly baseUrl = `${environment.apiUrl}/pagos`;
  private readonly recibosUrl = `${environment.apiUrl}/recibos`;

  constructor(private readonly http: HttpClient) {}

  procesarPagoCuentas(request: ProcesarPagoRequest): Observable<ProcesarPagoResponse> {
    return this.http.post<ProcesarPagoResponse>(`${this.baseUrl}/procesar-cuentas`, request);
  }

  canjeBancario(request: CanjeBancarioRequest): Observable<Recibo> {
    return this.http.post<Recibo>(`${this.baseUrl}/canje-bancario`, request);
  }

  registrarIngresoExterno(request: IngresoExternoRequest): Observable<Recibo> {
    return this.http.post<Recibo>(`${this.baseUrl}/ingreso-externo`, request);
  }

  listarRecibos(fecha?: string, tipo?: TipoRecibo): Observable<Recibo[]> {
    let params = new HttpParams();
    if (fecha) {
      params = params.set('fecha', fecha);
    }
    if (tipo) {
      params = params.set('tipo', tipo);
    }
    return this.http.get<Recibo[]>(this.recibosUrl, { params });
  }

  obtenerRecibo(id: number): Observable<Recibo> {
    return this.http.get<Recibo>(`${this.recibosUrl}/${id}`);
  }
}

File: backend/src/main/java/com/cibertec/sigac/service/PagoService.java
package com.cibertec.sigac.service;

import java.time.LocalDate;
import java.util.List;

import com.cibertec.sigac.dto.CanjeBancarioRequest;
import com.cibertec.sigac.dto.IngresoExternoRequest;
import com.cibertec.sigac.dto.ProcesarPagoRequest;
import com.cibertec.sigac.dto.ProcesarPagoResponse;
import com.cibertec.sigac.dto.ReciboResponse;
import com.cibertec.sigac.entity.TipoRecibo;

public interface PagoService {

    ProcesarPagoResponse procesarPagoCuentas(ProcesarPagoRequest request);

    ReciboResponse canjeBancario(CanjeBancarioRequest request);

    ReciboResponse registrarIngresoExterno(IngresoExternoRequest request);

    List<ReciboResponse> listarRecibos(LocalDate fecha, TipoRecibo tipo);

    ReciboResponse obtenerReciboPorId(Long id);
}
